### PR TITLE
Add benchmarks for lua-capnproto

### PR DESCRIPTION
**Pull request by fsfod**
Added two benchmarks for lua-capnproto both encoding and decoding the benchmarks are based on the message definition used in lua-capnproto's benchmark but we feed it a bit more data.

### CODE REVIEW

File: benchmarks/capnproto_decode/bench.lua [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        s0 = {
+            f0 = 3.14,
+            f1 = 3.14159265358979,
+        },
+        l0 = {0, i, -i, 127 },
+        t0 = "hello",
+        e1 = "enum7",
+    }
+
+    -- Make the bool fields vary a bit to trigger side traces like real wold data.
```
Comment (ltratt) [inline benchmarks/capnproto_decode/bench.lua]: "wold" typo.

File: benchmarks/capnproto_encode/bench.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        s0 = {
+            f0 = 3.14,
+            f1 = 3.14159265358979,
+        },
+        l0 = {0, i, -i, 127 },
+        t0 = "hello",
+        e1 = "enum7",
+    }
+
+    -- Make the bool fields vary a bit to trigger side traces like real wold data.
```
Comment (ltratt) [inline benchmarks/capnproto_encode/bench.lua]: Ditto.

**Comment by ltratt** [comment]
Do files generated by capnproto have some sort of associated license?

File: benchmarks/capnproto_decode/bench.lua [hunk 3]
```diff
@@ -0,0 +1,50 @@
+local capnp = require "capnp"
+local ffi = require "ffi"
+local example_capnp = require "example_capnp"
+
+local list_size = 100
+local list = {}
+
+for i=1,list_size do
```
Comment (vext01) [inline benchmarks/capnproto_decode/bench.lua]: further down in the file, you have spaces after your commas. I suggest you choose one and stick with it.

File: benchmarks/capnproto_decode/bench.lua [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+local binary_string = example_capnp.T1List.serialize(t1list)
+assert(example_capnp.T1List.calc_size(t1list) == #binary_string)
+
+local result
+local parse = example_capnp.T1List.parse
+
+function run_iter(n)
+    for n=1,n do
+        result = parse(binary_string)
+        assert(#result.list == list_size)
```
Comment (vext01) [inline benchmarks/capnproto_decode/bench.lua]: So you only check the size of the data, not that it matches?

Comment (vext01) [reply benchmarks/capnproto_decode/bench.lua]: I guess it depends how much "checking" we want to do.

File: benchmarks/capnproto_encode/bench.lua [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+local t1list = { list = list }
+
+local size = example_capnp.T1List.calc_size(t1list)
+local buf = ffi.new("char[?]", size)
+local serialize = example_capnp.T1List.serialize
+
+function run_iter(n)
+    for n=1,n do
+        serialize(t1list, buf, size)
```
Comment (vext01) [inline benchmarks/capnproto_encode/bench.lua]: Did you want to check anything here?

Comment (fsfod) [reply benchmarks/capnproto_encode/bench.lua]: Its just a huge blob of binary data

**Comment by fsfod** [comment]
> Do files generated by capnproto have some sort of associated license?
Well its licensed under MIT as well as Cap’n Proto and but doesn't mention anything about generated files.

File: benchmarks/capnproto_encode/bench.lua (hunk 2 above)
Comment (fsfod) [reply benchmarks/capnproto_encode/bench.lua]: fixed

File: benchmarks/capnproto_decode/bench.lua (hunk 1 above)
Comment (fsfod) [reply benchmarks/capnproto_decode/bench.lua]: fixed

File: benchmarks/capnproto_decode/bench.lua (hunk 4 above)
Comment (fsfod) [reply benchmarks/capnproto_decode/bench.lua]: I was only sanity checking that it was producing some data.

**Comment by ltratt** [comment]
OK, that's fine -- thanks for checking.

File: benchmarks/capnproto_encode/bench.lua [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+local t1list = { list = list }
+
+local size = example_capnp.T1List.calc_size(t1list)
+local buf = ffi.new("char[?]", size)
+local serialize = example_capnp.T1List.serialize
+
+function run_iter(n)
+    for n=1, n do
+        serialize(t1list, buf, size)
```
Comment (vext01) [inline benchmarks/capnproto_encode/bench.lua]: OK, so do you want to check this call makes some data too then?

Comment (fsfod) [reply benchmarks/capnproto_encode/bench.lua]: I thought it didn't return anything but its only using the buffer as temp storage before turning it in to a string that it returns. so I will check the length at least.

**Comment by fsfod** [comment]
@ltratt anything else?

**Comment by ltratt** [comment]
@vext01 Are you OK with everything here?

**Comment by vext01** [comment]
LGTM

**Comment by ltratt** [comment]
@fsfod Ok, so if you squash then I can merge. Thanks!

**Comment by fsfod** [comment]
@ltratt squashed and ready to merge

**Pull request merged**